MoreLinq/TakeLast.cs

```diff
@@ -49,9 +49,11 @@ public static IEnumerable<TSource> TakeLast<TSource>(this IEnumerable<TSource> s
         {
             if (source == null) throw new ArgumentNullException(nameof(source));
 
-            return 
+            return
                 source is ICollection<TSource> col
                 ? col.Slice(Math.Max(0, col.Count - count), int.MaxValue)
+                : source is IReadOnlyCollection<TSource> readOnlyCol
+                ? readOnlyCol.Slice(Math.Max(0, readOnlyCol.Count - count), int.MaxValue)
```

Comment:
`col.Slice(Math.Max(0, col.Count - count), int.MaxValue)` reads identical to `readOnlyCol.Slice(Math.Max(0, readOnlyCol.Count - count), int.MaxValue)` and so the two should be consolidated via, perhaps, a local function.